web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileQuarantinesBlock.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dbcdk_community\Plugin\Block\ProfileQuarantinesBlock.
+ */
+
+namespace Drupal\dbcdk_community\Plugin\Block;
+
+use DBCDK\CommunityServices\Api\ProfileApi;
+use DBCDK\CommunityServices\Api\QuarantineApi;
+use DBCDK\CommunityServices\ApiException;
+use DBCDK\CommunityServices\Model\Quarantine;
+use Drupal\Core\Datetime\DateFormatter;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Link;
+use Drupal\Component\Utility\Xss;
+
+/**
+ * Provides a 'ProfileQuarantinesBlock' block.
+ *
+ * This block provides an overview of a Community Profile's Quarantines.
+ *
+ * @Block(
+ *   id = "dbcdk_community_profile_quarantines_block",
+ *   admin_label = @Translation("DBCDK Community Profile Quarantines"),
+ *   context = {
+ *     "username" = @ContextDefinition("string")
```

Comment:
Getting Profiles by username is a reoccuring thing but it seems to make some things a bit more complex e.g.

- Gettings profiles where we use `profileFindOne` with filter instead of `profileFindById`
- Generating links to quarantines where we have to pass the username along instead of using the `quarantinedId` already available on the `Quarantine` object
- Are we certain that the characters in usernames are enforced to be suitable url arguments?